src/exception.hh

```diff
@@ -23,6 +23,30 @@ private:
     String m_what;
 };
 
+struct key_parse_error : runtime_error
```

Comment:
This does not belong to exception.hh, which does not contain any specific option type. it should be directly in keys.cc as nothing else is going to know about it except the unit test (and I am still on the fence that we really need to introduce an exception for that, so please keep it to a single one)